src/Native/Runtime/AsmOffsets.h

```diff
@@ -4,179 +4,114 @@
 //
 
 //
-// This file is used by AsmOffsets.cpp to validate that our 
+// This file is used by AsmOffsets.cpp to validate that our
 // assembly-code offsets always match their C++ counterparts.
-//
+
 // You must #define PLAT_ASM_OFFSET and PLAT_ASM_SIZEOF before you #include this file
-//
 
+#ifdef BIT64
+#define ASM_OFFSET(offset32, offset64, cls, member) PLAT_ASM_OFFSET(offset64, cls, member)
+#define ASM_SIZEOF(sizeof32, sizeof64, cls        ) PLAT_ASM_SIZEOF(sizeof64, cls)
+#define ASM_CONST(const32, const64, expr)           PLAT_ASM_CONST(const64, expr)
+#else
+#define ASM_OFFSET(offset32, offset64, cls, member) PLAT_ASM_OFFSET(offset32, cls, member)
+#define ASM_SIZEOF(sizeof32, sizeof64, cls        ) PLAT_ASM_SIZEOF(sizeof32, cls)
+#define ASM_CONST(const32, const64, expr)           PLAT_ASM_CONST(const32, expr)
+#endif
+
+// NOTE: the values MUST be in hex notation WITHOUT the 0x prefix
+
+//        32-bit,64-bit, constant symbol
+ASM_CONST( 14c08, 14c08, RH_LARGE_OBJECT_SIZE)
+ASM_CONST(   400,   800, CLUMP_SIZE)
+ASM_CONST(     a,     b, LOG2_CLUMP_SIZE)
+
+//        32-bit,64-bit, class, member
+ASM_OFFSET(    0,     0, Object, m_pEEType)
+
+ASM_OFFSET(    4,     8, Array, m_Length)
+
+ASM_OFFSET(    0,     0, EEType, m_usComponentSize)
+ASM_OFFSET(    2,     2, EEType, m_usFlags)
+ASM_OFFSET(    4,     4, EEType, m_uBaseSize)
+ASM_OFFSET(   14,    18, EEType, m_VTable)
+
+ASM_OFFSET(    0,     0, Thread, m_rgbAllocContextBuffer)
+ASM_OFFSET(   1c,    28, Thread, m_ThreadStateFlags)
+ASM_OFFSET(   20,    30, Thread, m_pTransitionFrame)
+ASM_OFFSET(   24,    38, Thread, m_pHackPInvokeTunnel)
+ASM_OFFSET(   34,    58, Thread, m_ppvHijackedReturnAddressLocation)
+ASM_OFFSET(   38,    60, Thread, m_pvHijackedReturnAddress)
+ASM_OFFSET(   3c,    68, Thread, m_pExInfoStackHead)
+
+ASM_SIZEOF(   14,    20, EHEnum)
+
+ASM_OFFSET(    0,     0, alloc_context, alloc_ptr)
+ASM_OFFSET(    4,     8, alloc_context, alloc_limit)
+
+ASM_OFFSET(    4,     8, RuntimeInstance, m_pThreadStore)
+
+#ifdef FEATURE_CACHED_INTERFACE_DISPATCH
+ASM_OFFSET(    4,     8, InterfaceDispatchCell, m_pCache)
+ASM_OFFSET(   10,    20, InterfaceDispatchCache, m_rgEntries)
```

Comment:
I agree, I hesitated about this m_pCell field. Is the conditional definition of that field really necessary since it is part of a union which already has a member so always having it defined on _AMD64_ would not hurt?